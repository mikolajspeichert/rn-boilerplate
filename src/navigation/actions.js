export default {
  login: {
    MOVE: 'LOGIN_MOVE',
  },
  main: {
    MOVE: 'MAIN_MOVE',
  },
}
