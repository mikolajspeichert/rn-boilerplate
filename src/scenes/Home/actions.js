import realm from '@storage'

const LOAD_ITEMS = 'LOAD_ITEMS'

export const actions = {
  LOAD_ITEMS,
}

const loadItems = async () => {
//  const items = await
}
