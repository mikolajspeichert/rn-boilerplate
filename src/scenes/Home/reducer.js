var initialState = {
  items: [],
  theme: '#fafafa',
}

export default (state = initialState, action) => {
  switch (action.type) {
    default:
      return state
  }
}
