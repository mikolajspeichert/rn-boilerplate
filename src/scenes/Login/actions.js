import NavigatorActions from '@navigation/actions'

const MOVE = 'MOVE'

export const actions = {
  MOVE,
}

export const moveToOnboarding = () => {
  return {
    type: NavigatorActions.login.MOVE,
    value: 'Onboarding',
  }
}
