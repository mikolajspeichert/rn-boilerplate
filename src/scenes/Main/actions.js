import NavigatorActions from '@navigation/actions'

const DEFAULT_ACTION = 'DEFAULT_ACTION'

export const actions = {
  DEFAULT_ACTION,
}

export const defaultAction = () => {
  return async dispatch => {
    dispatch({
      type: DEFAULT_ACTION,
      value: [],
    })
  }
}

export const moveToAdd = () => {
  return {
    type: NavigatorActions.MOVE,
    value: 'Add',
  }
}
export const moveToSettings = () => {
  return {
    type: NavigatorActions.MOVE,
    value: 'Settings',
  }
}
export const moveToLogin = () => {
  return {
    type: NavigatorActions.MOVE,
    value: 'Login',
  }
}
