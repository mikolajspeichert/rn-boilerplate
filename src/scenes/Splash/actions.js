import NavigatorActions from '@navigation/actions'

const DEFAULT_ACTION = 'DEFAULT_ACTION'

export const actions = {
  DEFAULT_ACTION,
}

export const moveToMain = () => {
  return async dispatch => {
    setTimeout(() => {
      dispatch({
        type: NavigatorActions.MOVE,
        value: 'Main',
      })
    }, 1000)
  }
}

export const moveToLogin = () => {
  return async dispatch => {
    setTimeout(() => {
      dispatch({
        type: NavigatorActions.MOVE,
        value: 'Login',
      })
    }, 1000)
  }
}
