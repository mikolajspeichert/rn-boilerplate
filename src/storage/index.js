import Realm from 'realm'
import Item from './item'

export default new Realm({ schema: [Item] })
