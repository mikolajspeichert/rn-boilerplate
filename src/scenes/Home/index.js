import React, { Component } from 'react'
import { View, Text, Image, Platform, Button } from 'react-native'
import { connect } from 'react-redux'
import { createDefaultNotification } from '@services/notification'
import icons from '@assets/icons'
import styles from './styles'
import { loadItems } from './actions'
import I18n from '@i18n'

class HomeScreen extends Component {
  static navigationOptions = {
    title: I18n.t('home.hello'),
    tabBarIcon: <Image source={icons.check} style={styles.icon} />,
  }

  componentDidMount() {
    this.props.load()
  }

  render() {
    return (
      <View style={styles.home}>
        <Text>
          {I18n.t('home.welcome')} {Platform.OS === 'ios' ? 'iOS' : 'Android'}
        </Text>
        <Button onPress={() => createDefaultNotification()} title="Notify" />
      </View>
    )
  }
}
const mapStateToProps = state => {
  return {
    items: !!state.items && Object.values(state.items),
  }
}

const mapDispatchToProps = dispatch => ({
  goToMore: () => dispatch({ type: 'Add' }),
  load: () => dispatch(loadItems()),
})

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen)
