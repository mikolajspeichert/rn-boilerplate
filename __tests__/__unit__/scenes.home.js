import 'react-native'
import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { shallow } from 'enzyme'
// import sinon from 'sinon'
import HomeScreen from '@scenes/Home'
import { loadItems, addItem } from '@scenes/Home/actions'
import storage from '@storage'

const state = {}
const mockStore = configureStore([thunk])

beforeAll(async () => {
  let realm = await storage
  realm.write(() => {
    realm.create('Item', {
      id: 'first',
      title: 'FIRST_ITEM',
      notes: 'No notes on this one',
      created: new Date(1),
    })
    realm.create('Item', {
      id: 'second',
      title: 'Second item',
      notes: 'Some notes',
      created: new Date(100),
    })
  })
})

afterAll(async () => {
  let realm = await storage
  realm.write(() => {
    realm.delete(realm.objects('Item'))
  })

  realm.close()
})

describe('Home Screen', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<HomeScreen store={mockStore(state)} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('dispatches navigation action', () => {
    console.log()
  })
})

describe('Home action', () => {
  it('loads items correctly', done => {
    const mockDispatch = action => {
      expect(action).toMatchSnapshot()
      done()
    }
    loadItems()(mockDispatch)
  })

  it('adds item correctly', done => {
    const mockItem = {
      id: 'mock',
      title: 'mock',
    }
    const mockDispatch = action => {
      expect(action).toMatchSnapshot()
      done()
    }
    addItem(mockItem)(mockDispatch)
  })
})
