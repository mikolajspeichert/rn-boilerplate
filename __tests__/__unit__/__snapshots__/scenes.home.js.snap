// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(HomeScreen)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "goToMore": [Function],
      "items": false,
      "load": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "goToMore": [Function],
        "items": false,
        "load": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Home action adds item correctly 1`] = `
Object {
  "type": "TRY_ADD_ITEM",
  "value": Object {
    "id": "mock",
    "title": "mock",
  },
}
`;

exports[`Home action loads items correctly 1`] = `
Object {
  "type": "LOAD_ITEMS",
  "value": Results {
    "0": Item {
      "created": 1970-01-01T00:00:00.001Z,
      "id": "first",
      "notes": "No notes on this one",
      "title": "FIRST_ITEM",
    },
    "1": Item {
      "created": 1970-01-01T00:00:00.100Z,
      "id": "second",
      "notes": "Some notes",
      "title": "Second item",
    },
  },
}
`;
